### ISSUE TITLE
Interactive Spyql

### ISSUE THREAD

**Issue by yashbonde** [dcmoura/spyql#64]
Hi,

I am curious to see if we can implement interactive spyql that can be called from within any script. Querying python objects is something pretty powerful that should be brought directly into the python language like how C# has LINQ. I have started to work a bit on it and here's some pseudo code I had in mind:

```python
class Q:
  def __init__(self, query):
    self._prs, self._string_regex = parse(clean_query(query))
  
  def __call__(self, **kwargs):
    pass
```

And usage can be as follows:

```python
>>> q = Q("IMPORT numpy SELECT numpy.mean(data->salary) FROM data WHERE data->name == 'akash'")
>>> q(data = data)
```

Though I can hack it as you have done in the `main_test.py` by getting a `CliRunner` but that seems unintuitive when using interactively. I'd like to know your thoughts on this and some pointers to start working on it!

**Comment by dcmoura** [comment]
Hi @yashbonde, that's interesting! 

I must say that somewhere along the way I decided to program spyql as a CLI and not as as a lib. Didn't think people would like the idea of querying objects with (SQL) strings...
That being said, I am currently planning an implementation of an IPython magic command for running spyql queries on Jupyter notebooks, and that might require some refactoring to make spyql easily available as a lib where you could pass variables, options, etc.  I am also thinking on sharing (by default) a brother scope of variables (e.g. share locals/globals vars).  

Using the CliRunner is not intuitive at all... we really need to work on having a better API :-) 

Going forward with your implementation of `Q`, I would suggest 
1. calling `Processor.go()` in `Q.__call()` like done here: https://github.com/dcmoura/spyql/blob/c379afa89c883b29bc4d3eaf3eea5a7c9d9cbd91/spyql/cli.py#L338
2. adding an extra argument to `Processor.go()` to allow passing a dictionary of variables `my_vars`
3. tweaking the following call to pass `my_vars` when initialising the variables of the spyql query: https://github.com/dcmoura/spyql/blob/c379afa89c883b29bc4d3eaf3eea5a7c9d9cbd91/spyql/processor.py#L310
and here: https://github.com/dcmoura/spyql/blob/c379afa89c883b29bc4d3eaf3eea5a7c9d9cbd91/spyql/processor.py#L22

Bottom line, currently you would need to fork/clone and tweak the code, but I am planning on implementing this soon. 
Let me know if would like to jump in and help with the refactoring :-)

**Comment by yashbonde** [comment]
Sounds good, I'll try to implement your suggestions, come up with some PoC and we can see from there. I am not aware on how to program magic methods (I don't really use them :P), but once this is done I'd like to help out more.
